pkg/streamingpromql/testdata/ours/functions.test

```diff
@@ -682,3 +682,22 @@ eval range from 0 to 7m step 1m last_over_time(some_metric_count[3m])
   some_metric_count{env="prod", cluster="eu"} _ _ _ 0 1 2 3 4
   some_metric_count{env="prod", cluster="us"} _ _ _ 0 2 4 6 8
   some_metric_count{env="prod", cluster="au"} _ _ _ {{count:5}} {{count:10}} {{count:15}} {{count:20}} {{count:25}}
+
+clear
+
+load 30s
+  some_metric{case="floats only"} 1 2 _ 3 4
+  some_metric{case="histograms only"} {{count:1}} {{count:2}} _ {{count:3}} {{count:4}}
+  some_metric{case="floats and histograms"} 1 2 _ {{count:3}} {{count:4}}
+
+# If timestamp() is used over a selector, it returns the timestamps of the underlying samples.
+eval range from 0 to 2m step 1m timestamp(some_metric)
+  {case="floats only"} 0 30 120
+  {case="histograms only"} 0 30 120
+  {case="floats and histograms"} 0 30 120
+
+# If timestamp() is used over something that is not a selector, it returns the timestamps of the points returned.
```

Comment:
I think it'd be nice to see a test for something like `timestamp(rate(metric[5m]))`  or similar